app/scripts/models/user.js

```diff
@@ -382,6 +392,27 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Remove accounts with empty or `undefined` uids.
+     * See #4769. w/ e10s enabled, post account reset,
+     * a phantom account with a uid of `undefined` was
+     * being written to localStorage. These accounts
+     * are garbage, get rid of them.
+     *
+     * @returns {Promise}
+     */
+    removeAccountsWithInvalidUid () {
+      return p().then(() => {
+        const accounts = this._accounts();
+        for (const uid in accounts) {
+          if (! uid || uid === 'undefined') {
```

Comment:
Yup, accounts are being written with the string `undefined`. I'll update the comments for clarity.